From 50170c82ac2798a64afd20e70a18990c14365d01 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 21 Jun 2018 11:39:39 +0800
Subject: [PATCH 4001/5242] MLK-18649-3 arm64: dts: fsl-imx8qxp-mek: Supply
 aux PD for LVDS dual chan mode

commit  ca5d2f29dce9ef97378dd80c22a660d7830ed161 from
https://source.codeaurora.org/external/imx/linux-imx.git

When ldb works in dual LVDS channel mode, it consumes two MIPI/LVDS combo
subsystems, one primary and the other auxiliary.  Each subsystem is powered
by it's own power domain.  So, the dual channel mode needs two power domains.
The pd_mipi_dsi_0/1_dual_lvds power domains may supply two power domains to
the ldb1/2 nodes respectively.  Thus, let's hook them to the ldb1/2 nodes
when ldb1/2 works in dual channel mode.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
(cherry picked from commit 12dd916ad6bd4f10bf41e27e4392787b4a3af0e5)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../fsl-imx8qxp-mek-it6263-lvds0-dual-channel.dts  |    1 +
 .../fsl-imx8qxp-mek-it6263-lvds1-dual-channel.dts  |    1 +
 .../fsl-imx8qxp-mek-jdi-wuxga-lvds0-panel.dts      |    1 +
 .../fsl-imx8qxp-mek-jdi-wuxga-lvds1-panel.dts      |    1 +
 4 files changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-it6263-lvds0-dual-channel.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-it6263-lvds0-dual-channel.dts
index 06ec204..aa6d7b2 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-it6263-lvds0-dual-channel.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-it6263-lvds0-dual-channel.dts
@@ -22,6 +22,7 @@
 
 &ldb1 {
 	fsl,dual-channel;
+	power-domains = <&pd_mipi_dsi_0_dual_lvds>;
 };
 
 &ldb2 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-it6263-lvds1-dual-channel.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-it6263-lvds1-dual-channel.dts
index 24a8d56..14328ef 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-it6263-lvds1-dual-channel.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-it6263-lvds1-dual-channel.dts
@@ -26,4 +26,5 @@
 
 &ldb2 {
 	fsl,dual-channel;
+	power-domains = <&pd_mipi_dsi_1_dual_lvds>;
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-jdi-wuxga-lvds0-panel.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-jdi-wuxga-lvds0-panel.dts
index b87fc07..f855f66 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-jdi-wuxga-lvds0-panel.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-jdi-wuxga-lvds0-panel.dts
@@ -29,6 +29,7 @@
 
 &ldb1 {
 	fsl,dual-channel;
+	power-domains = <&pd_mipi_dsi_0_dual_lvds>;
 
 	lvds-channel@0 {
 		fsl,data-mapping = "spwg";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-jdi-wuxga-lvds1-panel.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-jdi-wuxga-lvds1-panel.dts
index 1e1bb1c..b23fb2b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-jdi-wuxga-lvds1-panel.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-jdi-wuxga-lvds1-panel.dts
@@ -29,6 +29,7 @@
 
 &ldb2 {
 	fsl,dual-channel;
+	power-domains = <&pd_mipi_dsi_1_dual_lvds>;
 
 	lvds-channel@0 {
 		fsl,data-mapping = "spwg";
-- 
1.7.9.5

